src/fhirtypes/ElementDefinition.ts

```diff
@@ -1520,12 +1520,22 @@ export class ElementDefinition {
    * @throws {ValueAlreadyAssignedError} when the code is already assigned to a different code
    * @throws {InvalidUriError} when the system being assigned is not a valid uri
    */
-  private assignFshCode(code: FshCode, exactly = false): void {
+  private assignFshCode(code: FshCode, fisher?: Fishable, exactly = false): void {
     if (code.system && !isUri(code.system.split('|')[0])) {
       throw new InvalidUriError(code.system);
     }
 
     const type = this.type[0].code;
+
+    // Check if the code is a specialization of the Quantity type
+    let quantityType = type === 'Quantity';
+    if (!quantityType) {
+      const actualTypeSD = fisher?.fishForFHIR(type, Type.Type);
+      if (actualTypeSD?.baseDefinition === 'http://hl7.org/fhir/StructureDefinition/Quantity') {
+        quantityType = true;
+      }
+    }
```

Comment:
Since this is executed _before_ the `if`/`else if` checks below, it's being executed even in the case that the type is `code`, `string`, `uri`, `Coding`, or `CodeableConcept`.  Since _fishing_ can be an expensive operation, and since the quantity specialization use case is the minority use case (meaning this would happen needlessly in most cases), I think it would be best to delay this until we know it's not one of those other cases.

One potential approach would be to create a tiny helper function based on your code above:
```ts
private isQuantityType(type: string, fisher?: Fishable): boolean {
    if (type === 'Quantity') {
      return true;
    }
    const actualTypeSD = fisher?.fishForFHIR(type, Type.Type);
    return actualTypeSD?.baseDefinition === 'http://hl7.org/fhir/StructureDefinition/Quantity';
  }
}
```
Then in the `else if` (line 1550), you would do:
```ts
} else if (isQuantityType(type, fisher)) {
```
this would delay execution of the moderately expensive code until we _know_ we need to check for a quantity type.

Reply:
I really like this solution, and it'd be much more efficient as you were saying. I'll implement this approach and switch the parameters back in a new commit.